Sources/SwifterSwift/UIKit/UIButtonExtensions.swift

```diff
@@ -167,34 +167,72 @@ public extension UIButton {
-                let text = titleLabel?.text,
-                let font = titleLabel?.font else { return }
-
-            let titleSize = text.size(withAttributes: [.font: font])
-
-            let titleOffset = -(imageSize.height + spacing)
-            titleEdgeInsets = UIEdgeInsets(top: 0.0, left: -imageSize.width, bottom: titleOffset, right: 0.0)
-
-            let imageOffset = -(titleSize.height + spacing)
-            imageEdgeInsets = UIEdgeInsets(top: imageOffset, left: 0.0, bottom: 0.0, right: -titleSize.width)
-
-            let edgeOffset = abs(titleSize.height - imageSize.height) / 2.0
-            contentEdgeInsets = UIEdgeInsets(top: edgeOffset, left: 0.0, bottom: edgeOffset, right: 0.0)
-        } else {
-            let insetAmount = spacing / 2
-            imageEdgeInsets = UIEdgeInsets(top: 0, left: -insetAmount, bottom: 0, right: insetAmount)
-            titleEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: -insetAmount)
-            contentEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: insetAmount)
-        }
+    func setImageLeftOfText(spacing: CGFloat) {
+        let insetAmount = spacing / 2
+        imageEdgeInsets = UIEdgeInsets(top: 0, left: -insetAmount, bottom: 0, right: insetAmount)
+        titleEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: -insetAmount)
+    }
+
+    /// SwifterSwift: Set the image to the right of the text
+    ///
+    /// - Parameters:
+    ///   - spacing: spacing between title text and image.
+    func setImageRightOfText(spacing: CGFloat) {
+        // https://stackoverflow.com/questions/2451223/#7199529
+        guard
+            let imageSize = imageView?.image?.size,
+            let text = titleLabel?.text,
+            let font = titleLabel?.font else { return }
+
+        let titleSize = text.size(withAttributes: [.font: font])
+
+        let titleOffset = -(imageSize.width + spacing / 2)
+        titleEdgeInsets = UIEdgeInsets(top: 0, left: titleOffset, bottom: 0, right: -titleOffset)
+
+        let imageOffset = -(titleSize.width + spacing / 2)
+        imageEdgeInsets = UIEdgeInsets(top: 0, left: -imageOffset, bottom: 0, right: imageOffset)
+    }
+
+    /// SwifterSwift: Set the image above the text
+    ///
+    /// - Parameters:
+    ///   - spacing: spacing between title text and image.
+    func setImageAboveText(spacing: CGFloat) {
+        guard
+            let imageSize = imageView?.image?.size,
+            let text = titleLabel?.text,
+            let font = titleLabel?.font else { return }
+
+        let titleSize = text.size(withAttributes: [.font: font])
+
+        let titleOffset = -(imageSize.height + spacing)
+        titleEdgeInsets = UIEdgeInsets(top: 0.0, left: -imageSize.width, bottom: titleOffset, right: 0.0)
+
+        let imageOffset = -(titleSize.height + spacing)
+        imageEdgeInsets = UIEdgeInsets(top: imageOffset, left: 0.0, bottom: 0.0, right: -titleSize.width)
+    }
```

Comment:
```suggestion
    func setImageAboveText(spacing: CGFloat) {
        // https://stackoverflow.com/questions/2451223/#7199529
        guard
            let imageSize = imageView?.image?.size,
            let text = titleLabel?.text,
            let font = titleLabel?.font else { return }

        let titleSize = text.size(withAttributes: [.font: font])

        let titleOffset = -(imageSize.height + spacing)
        titleEdgeInsets = UIEdgeInsets(top: 0.0, left: -imageSize.width, bottom: titleOffset, right: 0.0)

        let imageOffset = -(titleSize.height + spacing)
        imageEdgeInsets = UIEdgeInsets(top: imageOffset, left: 0.0, bottom: 0.0, right: -titleSize.width)

        let horizontalInsets = min(imageSize.width, titleSize.width) / -2
        let verticalInsets = (min(imageSize.height, titleSize.height) + spacing) / 2
        contentEdgeInsets = UIEdgeInsets(top: verticalInsets, left: horizontalInsets, bottom: verticalInsets, right: horizontalInsets)

        sizeToFit()
    }
```

Reply:
I think the above and below styles do not need to be processed, but the user needs to adjust the size of the buttons according to their needs.

Reply:
The insets have to be set correctly, including `contentEdgeInsets`, or the layout of the button is broken. The user can still set the size as they want, just remove `sizeToFit`.

Reply:
Test in project:

#### normal: 
<img width="279" alt="normal" src="https://user-images.githubusercontent.com/12928482/98759964-293d7380-240d-11eb-9e45-a01815eb13a1.png">

#### without contentEdgeInsets: 
<img width="266" alt="without contentEdgeInsets" src="https://user-images.githubusercontent.com/12928482/98760047-61dd4d00-240d-11eb-9ea2-fa36a33788ff.png">

#### set contentEdgeInsets:
<img width="280" alt="set contentEdgeInsets" src="https://user-images.githubusercontent.com/12928482/98760109-8a654700-240d-11eb-8b0d-04401441447b.png">

#### set sizeToFit:
<img width="285" alt="set sizeToFit" src="https://user-images.githubusercontent.com/12928482/98760152-a537bb80-240d-11eb-9570-ff215288adef.png">

#### set numberOfLines and contentEdgeInsets:
<img width="273" alt="set numberOfLines and contentEdgeInsets" src="https://user-images.githubusercontent.com/12928482/98760223-c8fb0180-240d-11eb-9d98-95a2d3faccdd.png">

#### set numberOfLines without contentEdgeInsets:
<img width="260" alt="set numberOfLines without contentEdgeInsets" src="https://user-images.githubusercontent.com/12928482/98760256-ddd79500-240d-11eb-8f08-41f895f6ddec.png">

The background color shows the size of the button. If the text is not too long, it will have no effect whether it is set or not set `contentEdgeInsets`.
As can be seen from the above, it meets our needs without setting `contentEdgeInsets`, If we have to set contentEdgeInsets, we have to find the correct setting.
In any case, we must first meet the needs of the project, not the playground.

Reply:
Can you please share the code from your project? Because when I test it, using autolayout or the unit tests, I don't have these problems.

Using all my suggested code:
![Simulator Screen Shot - iPhone 11 - 2020-11-11 at 09 53 21](https://user-images.githubusercontent.com/924764/98790936-a6181f80-2404-11eb-9deb-edabb1d27eb7.png)

Not setting contentEdgeInstets
![Simulator Screen Shot - iPhone 11 - 2020-11-11 at 09 54 00](https://user-images.githubusercontent.com/924764/98790974-b16b4b00-2404-11eb-8462-264799450018.png)

Not setting horizontal contentEdgeInsets
![Simulator Screen Shot - iPhone 11 - 2020-11-11 at 09 59 02](https://user-images.githubusercontent.com/924764/98791004-ba5c1c80-2404-11eb-9868-0fc459f1ad36.png)

Reply:
Your text is too short, you need to test with a very long text.